### PR TITLE
fix(core): amend typing for `jsonParse()` options

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

**Comment by netroy** [comment]
is there anything that we could cherry-pick from https://github.com/n8n-io/n8n/commit/1f87836f5dee7c584c1e7629fa4bff2f7ec1c570 ?

File: packages/workflow/src/utils.ts [hunk 1]
```diff
@@ -30,22 +30,25 @@ export const deepCopy = <T>(source: T): T => {
...
-type FallbackValue<T> = { fallbackValue: T };
 
-export const jsonParse = <T>(
-	jsonString: string,
-	options: ErrorMessage | FallbackValue<T> | {} = {},
-): T => {
+type MutuallyExclusiveTypeError = 'TypeError: Only one of the two options is allowed';
+
+type MutuallyExclusive<T, U> =
+	| (T & { [k in Exclude<keyof U, keyof T>]?: MutuallyExclusiveTypeError })
```
Comment (netroy) [inline packages/workflow/src/utils.ts:37]: I prefer `never` over `MutuallyExclusiveTypeError`, as the error thrown by `tsc` is equally confusing with either option, but with `never` you don't need to force case `fallbackValue`.

Comment (ivov) [reply packages/workflow/src/utils.ts:37]: I decided to accept the assertion in return for being able to immediately flag this when introducing the error, where it is likely to be seen. If only we had custom type errors in TS. I'll undo.

image

**Review by netroy — APPROVED** [review]
👍🏽

File: packages/workflow/src/utils.ts (hunk 1 above)
Comment (netroy) [reply packages/workflow/src/utils.ts]: we should revisit stuff like this at some point. I like the idea of explicit and clear compile-time errors.

**Comment by janober** [comment]
Got released with `n8n@0.200.0`

**Pull request merged**